### PR TITLE
match_name() can now take ald_scenario_demo

### LINKED ISSUE #195: Adapt r2dii.match workflow to be robust to `ald_demo` or `ald_scenario_demo` input
In anticipation for 2DegreesInvesting/r2dii.data#48, we need to adapt `r2dii.match` to be robust to input data using either `ald_demo` or `ald_scenario_demo`. 

I will play around with the code today and record my findings of what seems to break here.

### PR DESCRIPTION
**Pull request by maurolepore**
Closes #195 

@jdhoffa, the commits are quite small and should reflect the refactoring moves I used. My only problem is that I mindlessly duplicated all tests to ensure they work just as well with both `ald_demo` and `ald_scenario_demo`. There is only one thoughtful test. If you come up with more, please add them.

### CODE REVIEW

File: R/match_name2.R [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#     "ald_timestamp"
+#   )) %>%
+#   mutate(id_name = "company_name", .before = 1L) %>%
+#   mutate(
+#     ald_emission_factor_unit = glue::glue("{ald_sector} emission_factor"),
+#     .after = ald_emission_factor
+#   ) %>%
+#   inner_join(scenario_demo_2020_with_source, by = scenario_columns()) %>%
+#   pick_ald_location_in_region() %>%
+#   rename(scenario_region = region)
```
Comment (maurolepore) [inline R/match_name2.R]: Forget about this. It should never have been here.

**Comment by jdhoffa** [comment]
https://github.com/2DegreesInvesting/r2dii.match/blob/6784e92a81c438be6fb5f61fc65c48423e3e6b69/R/match_name.R#L13-L14
Should we update this to reference either `ald_demo` or `ald_scenario_demo` (or should we define an `ald` family of objects?)

**Comment by maurolepore** [comment]
I see errors on CI that I need to explore further..

**Comment by jdhoffa** [comment]
The error comes from the fact that `ald_scenario_demo`is only available in the latest remote version of `r2dii.data` so all tests touching `ald_scenario_demo` fail

**Comment by maurolepore** [comment]
Thanks, we then need to `use_dev_package("r2dii.data")` until we publish r2dii.data 0.0.4 on CRAN

**Comment by maurolepore** [comment]
Thanks @jdhoffa. 
I assume you continue with the review. If not let me know or directly merge.

**Pull request merged**